src/tcn_ext/tcn_keys.rs

```diff
@@ -1,80 +1,169 @@
-use crate::preferences::Preferences;
-use tcn::{TemporaryContactKey, ReportAuthorizationKey, MemoType, SignedReport, Error};
-use std::{cmp, io::Cursor, sync::Arc};
+use crate::preferences::{Preferences, PreferencesMock, TckBytesWrapper, TCK_SIZE_IN_BYTES};
 use cmp::max;
+use std::{cmp, io::Cursor, sync::Arc};
+use tcn::{Error, MemoType, ReportAuthorizationKey, SignedReport, TemporaryContactKey};
 
 pub trait TcnKeys {
-  fn crate_report(&self, report: Vec<u8>) -> Result<SignedReport, Error>;
+    fn create_report(&self, report: Vec<u8>) -> Result<SignedReport, Error>;
 }
 
 pub struct TcnKeysImpl<PreferencesType: Preferences> {
-  pub preferences: Arc<PreferencesType>,
+    pub preferences: Arc<PreferencesType>,
 }
 
-impl <PreferencesType: Preferences> TcnKeys for TcnKeysImpl<PreferencesType> {
-  fn crate_report(&self, report: Vec<u8>) -> Result<SignedReport, Error> {
-    let end_index = self.tck().index();
-    let minutes_in_14_days: u32 = 14 * 24 * 60 * 60;
-    let periods = minutes_in_14_days / 15;
-    let start_index = max(0, (end_index as u32) - periods) as u16;
+impl<PreferencesType: Preferences> TcnKeys for TcnKeysImpl<PreferencesType> {
+    fn create_report(&self, report: Vec<u8>) -> Result<SignedReport, Error> {
+        let end_index = self.tck().index();
+        //Important: logic change
```

Comment:
Thanks for the fixes! This comment can be removed, as it's relevant only for the PR.

Reply:
removed